patches/minecraft/net/minecraft/entity/player/PlayerEntity.java.patch

```diff
@@ -54,6 +54,15 @@
     }
  
     protected boolean func_204229_de() {
+@@ -345,7 +355,7 @@
+          } else {
+             pose = Pose.STANDING;
+          }
+-
++         pose = net.minecraftforge.event.ForgeEventFactory.getPlayerPose(this,pose);
```

Comment:
As far as I am aware events are quite optimized and barely affect performance. Furthermore one tick event per player is negligible compared to the thousands of other events.
Still, you are right, something that does not have to be executed every tick would be better.

However, I fear it might cause inconsistencies if the player entity is recreated/the world is loaded and the forcePose isn't set yet. Also mods could forget to clear the pose.
Since vanilla checks the pose every tick, I think it would be best if mods also check their intended pose every tick

Main reason for this approach is the reduced patch size though. Instead of a new field, a setter and some logic, this is just one modified line

Reply:
I disagree that another tick event is negligible. It is also unnecessary, as the same thing can be accomplished with an AT for `Entity#setPose` and using the existing tick event (with phase POST).
The argument for forgetting to clear it is valid, but I would consider it a mod bug, and therefore not to responsibility of forge to resolve it.